music_assistant/server/providers/tidal/__init__.py

```diff
@@ -0,0 +1,390 @@
+    library_items_add_remove,
+    parse_album,
+    parse_artist,
+    parse_playlist,
+    parse_track,
+    search,
+    tidal_code_login,
+    tidal_session,
+)
+
+if TYPE_CHECKING:
+    from music_assistant.common.models.config_entries import ProviderConfig
+    from music_assistant.common.models.provider import ProviderManifest
+    from music_assistant.server import MusicAssistant
+    from music_assistant.server.models import ProviderInstanceType
+
+TOKEN_TYPE = "Bearer"
+CONF_ACTION_AUTH = "auth"
+CONF_AUTH_TOKEN = "auth_token"
+CONF_REFRESH_TOKEN = "refresh_token"
+CONF_USER_ID = "user_id"
+CONF_EXPIRY_TIME = "expiry_time"
+TIDAL_SESSION = None
+
+
+async def setup(
+    mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig
+) -> ProviderInstanceType:
+    """Initialize provider(instance) with given configuration."""
+    prov = TidalProvider(mass, manifest, config)
+    await prov.handle_setup()
+    return prov
+
+
+async def get_config_entries(
+    mass: MusicAssistant,
+    instance_id: str | None = None,  # noqa: ARG001
+    action: str | None = None,
+    values: dict[str, ConfigValueType] | None = None,
+) -> tuple[ConfigEntry, ...]:
+    """
+    Return Config entries to setup this provider.
+
+    instance_id: id of an existing provider instance (None if new instance setup).
+    action: [optional] action key called from config entries UI.
+    values: the (intermediate) raw values for config entries sent with the action.
+    """
+
+    # config flow auth action/step (authenticate button clicked)
+    if action == CONF_ACTION_AUTH:
+        async with AuthenticationHelper(mass, values["session_id"]) as auth_helper:
+            TIDAL_SESSION = await tidal_code_login(auth_helper)
+            if not TIDAL_SESSION.check_login():
+                raise LoginFailed("Authentication to Tidal failed")
+            # set the retrieved token on the values object to pass along
+            values[CONF_AUTH_TOKEN] = TIDAL_SESSION.access_token
+            values[CONF_REFRESH_TOKEN] = TIDAL_SESSION.refresh_token
+            values[CONF_EXPIRY_TIME] = TIDAL_SESSION.expiry_time.isoformat()
+            values[CONF_USER_ID] = str(TIDAL_SESSION.user.id)
+
+    # return the collected config entries
+    return (
+        ConfigEntry(
+            key=CONF_AUTH_TOKEN,
+            type=ConfigEntryType.SECURE_STRING,
+            label="Authentication token for Tidal",
+            description="You need to link Music Assistant to your Tidal account.",
+            action=CONF_ACTION_AUTH,
+            action_label="Authenticate on Tidal.com",
+            value=values.get(CONF_AUTH_TOKEN) if values else None,
+        ),
+        ConfigEntry(
+            key=CONF_REFRESH_TOKEN,
+            type=ConfigEntryType.SECURE_STRING,
+            label="Refresh token for Tidal",
+            description="You need to link Music Assistant to your Tidal account.",
+            action_label="Authenticate on Tidal.com",
```

Comment:
```suggestion
            hidden=True,
```

Reply:
I tested this and if I set it as hidden the auth button also becomes hidden.
The other entries I will hide though.

Reply:
Yeah, only hide the entries you (ab)use for the refresh token etc. and you do not want to have visible in the UI

Reply:
(Ab)used entries hidden